
<launch>

    <arg name="camera1"     default="camera_1" />
    <arg name="camera2"     default="camera_2" />
    <arg name="env_camera"      default="env_camera" />
    <arg name="image1"      default="/rgb/image_raw" />
    <arg name="image2"      default="/rgb/image_raw" />
    <arg name="face1"       default="face_1" />
    <arg name="face2"       default="face_2" />

    <param name="$(arg face1)_description" command="$(find xacro)/xacro --inorder $(find freeplay_sandbox)/urdf/face.xacro name:=$(arg face1)_0" />
    <param name="$(arg face2)_description" command="$(find xacro)/xacro --inorder $(find freeplay_sandbox)/urdf/face.xacro name:=$(arg face2)_0" />
    <param name="sandtray_description" command="$(find xacro)/xacro --inorder
                                                $(find freeplay_sandbox)/urdf/sandtray.xacro
                                                camera1:=$(arg camera1)
                                                camera2:=$(arg camera2)
                                                env_camera:=$(arg env_camera)" />

    <arg name="bag" />

    <arg name="start" default="0" />


    <!-- Push down all topics/nodelets into "camera" namespace -->
    <group ns="$(arg camera1)">

        <node pkg="gazr" type="estimate" name="gazr" output="screen">
            <param name="face_model" value="$(find gazr)/shape_predictor_68_face_landmarks.dat" />
            <param name="prefix" value="$(arg face1)" />
            <param name="image_transport" value="compressed" />
            <remap from="image" to="rgb/image_raw"/>
        </node>

        <node pkg="freeplay_sandbox" type="sandtray_gaze_viz" name="sandtray_gaze_viz">
            <param name="face_frame" value="$(arg face1)_0" /> <!-- always consider the first detected face (ie face_0) -->
        </node>

    </group>

    <!-- Push down all topics/nodelets into "camera" namespace -->
    <group ns="$(arg camera2)">

        <node pkg="gazr" type="estimate" name="gazr" output="screen">
            <param name="face_model" value="$(find gazr)/shape_predictor_68_face_landmarks.dat" />
            <param name="prefix" value="$(arg face2)" />
            <param name="image_transport" value="compressed" />
            <remap from="image" to="rgb/image_raw"/>
        </node>

        <node pkg="freeplay_sandbox" type="sandtray_gaze_viz" name="sandtray_gaze_viz">
            <param name="face_frame" value="$(arg face2)_0" /> <!-- always consider the first detected face (ie face_0) -->
        </node>

    </group>

    <node pkg="rosbag" type="play" name="rosbag_play_freeplay_sandbox"
        args="$(arg bag) --clock --start $(arg start)" />

</launch>
